service/web/adminserver.go

```diff
@@ -254,6 +255,68 @@ func (s adminServer) MigrateSubscriptions(ctx context.Context, subscriptions *me
 	return &emptypb.Empty{}, nil
 }
 
+// normalizeTopicName returns the actual topic name used in message broker
+func denormalizeTopicName(name string) string {
+	return strings.ReplaceAll(name, "_", "/")
+}
+
+func (s adminServer) CleanupTopics(ctx context.Context, projects *metrov1.Projects) (*metrov1.Topics, error) {
+	logger.Ctx(ctx).Infow("received request to cleanup topics", "projects", projects.GetProjects())
+	tops := metrov1.Topics{}
+	for _, p := range projects.Projects {
+		subs, err := s.subscriptionCore.List(ctx, subscription.Prefix+p)
+		if err != nil {
+			logger.Ctx(ctx).Errorw("failed to fetch project subscriptions", "project", p)
+			continue
+		}
+		validTopics := make(map[string]bool, 0)
+		for _, sub := range subs {
+
+			for _, v := range sub.GetDelayTopics() {
+				validTopics[v] = true
+			}
+			validTopics[sub.GetDeadLetterTopic()] = true
+			validTopics[sub.GetRetryTopic()] = true
+			// validTopics[sub.GetSubscriptionTopic()] = true ###Internal topics are not used and hence up for deletion
```

Comment:
Why is the SubscriptionTopic up for deletion? Isn't this the primary topic of the subscription.

Reply:
Subscription topics are internal topics that were created for a re-arch but the re-arch plan was dropped. Primary topics are not subscriptions topics.